roles/lib_utils/library/glusterfs_check_containerized.py

```diff
@@ -0,0 +1,187 @@
+- name: glusterfs volumes check
+  glusterfs_check_containerized
+    oc_bin: "/usr/bin/oc"
+    oc_conf: "/etc/origin/master/admin.kubeconfig"
+    oc_namespace: "glusterfs"
+    cluster_name: "glusterfs"
+'''
+
+
+def fail(module, err):
+    """Fail on error"""
+    result = {'failed': True,
+              'changed': False,
+              'msg': err,
+              'state': 'unknown'}
+    module.fail_json(**result)
+
+
+def call_or_fail(module, call_args):
+    """Call subprocess.check_output and return utf-8 decoded stdout or fail"""
+    try:
+        # Must decode as utf-8 for python3 compatibility
+        res = subprocess.check_output(call_args).decode('utf-8')
+    except subprocess.CalledProcessError as err:
+        fail(module, str(err))
+    return res
+
+
+def get_valid_nodes(module, oc_exec, exclude_node):
+    """Return a list of pods that might actually be running"""
+    call_args = oc_exec + ['get', 'nodes']
+    res = call_or_fail(module, call_args)
+    valid_nodes = []
+    for line in res.split('\n'):
+        fields = line.split()
+        if not fields:
+            continue
+        if fields[0] != exclude_node and fields[1] == "Ready":
+            valid_nodes.append(fields[0])
+    if not valid_nodes:
+        fail(module,
+             'Unable to find suitable node in get nodes output: {}'.format(res))
+    return valid_nodes
+
+
+def select_pod(module, oc_exec, cluster_name, valid_nodes):
+    """Select a pod to attempt to run gluster commands on"""
+    call_args = oc_exec + ['get', 'pods', '-owide']
+    res = call_or_fail(module, call_args)
+    # res is returned as a tab/space-separated list with headers.
+    res_lines = res.split('\n')
+    pod_name = None
+    name_search = 'glusterfs-{}'.format(cluster_name)
+    res_lines = list(filter(None, res.split('\n')))
+
+    for line in res_lines[1:]:
+        fields = line.split()
+        if not fields:
+            continue
+        if name_search in fields[0]:
+            if fields[2] == "Running" and fields[6] in valid_nodes:
+                pod_name = fields[0]
+                break
+
+    if pod_name is None:
+        fail(module,
+             "Unable to find suitable pod in get pods output: {}".format(res))
+    else:
+        return pod_name
+
+
+def get_volume_list(module, oc_exec, pod_name):
+    """Retrieve list of active volumes from gluster cluster"""
+    call_args = oc_exec + ['exec', pod_name, '--', 'gluster', 'volume', 'list']
+    res = call_or_fail(module, call_args)
+    # This should always at least return heketidbstorage, so no need to check
+    # for empty string.
+    return list(filter(None, res.split('\n')))
+
+
+def check_volume_health_info(module, oc_exec, pod_name, volume):
+    """Check health info of gluster volume"""
+    call_args = oc_exec + ['exec', pod_name, '--', 'gluster', 'volume', 'heal',
+                           volume, 'info']
+    res = call_or_fail(module, call_args)
+    # Output is not easily parsed
```

Comment:
@jarrpa no machine readible output options for these gluster calls?

Reply:
The gluster cli interface leaves a lot to be desired here.  There's nothing documented for machine readable output.

Reply:
bummer, any ability to influence that in the future would be great